File: 2.0/java/calls/addCallBroadcastBatch.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.callstexts.model.CallRecipient;
import com.callfire.api.client.api.campaigns.model.request.AddBatchRequest;
import com.callfire.api.client.api.common.model.ResourceId;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient callfireClient = new CallfireClient("api_login", "api_password");
        CallRecipient recipient1 = new CallRecipient();
        recipient1.setPhoneNumber("12135551122");
        CallRecipient recipient2 = new CallRecipient();
        recipient2.setPhoneNumber("12135553434");
        CallRecipient recipient3 = new CallRecipient();
        recipient3.setPhoneNumber("12135558090");
        Map<String,String> attrs = new HashMap<>();
        attrs.put("custom_external_id", "30005044");
        attrs.put("custom_property_1", "value1");
        recipient3.setAttributes(attrs);

        AddBatchRequest request = AddBatchRequest.create()
            .campaignId(11646003L)
            .name("contact batch for call broadcast")
            .scrubDuplicates(true)
            .recipients(Arrays.asList(recipient1, recipient2, recipient3))
            //or you can add contacts from particular contact list
            //.contactListId(70055003L)
            .build();
        ResourceId resourceId = callfireClient.callBroadcastsApi().addBatch(request);
    }
}

File: 2.0/csharp/calls/addCallBroadcastBatch.cs
using System.Collections.Generic;
using CallfireApiClient;
using CallfireApiClient.Api.CallsTexts.Model;
using CallfireApiClient.Api.Campaigns.Model;
using CallfireApiClient.Api.Campaigns.Model.Request;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var request1 = new AddBatchRequest
        {
            CampaignId = 11646003,
            Name = "contact batch for call broadcast",
            Recipients = new List<Recipient>
            {
                new CallRecipient {PhoneNumber = "12135551122"},
                new CallRecipient {PhoneNumber = "12135553434"},
                new CallRecipient
                {
                    PhoneNumber = "12135558090",
                    Attributes = new Dictionary<string, string>
                    {
                        {"custom_external_id", "30005044"},
                        {"custom_property_1", "value1"}
                    }
                }
            },
            //or you can add contacts from particular contact list
            //ContactListId = 70055003
            ScrubDuplicates = true
        };
        var resourceId = client.TextBroadcastsApi.AddBatch(request1);
    }
}
